Mastodon/Scene/Notification/Notification Filtering/Requests/NotificationRequestsTableViewController.swift

```diff
@@ -0,0 +1,226 @@
+        }
+
+        dismissAction.image = UIImage(systemName: "speaker.slash")
+
+        let swipeAction = UISwipeActionsConfiguration(actions: [dismissAction])
+        swipeAction.performsFirstActionWithFullSwipe = true
+        return swipeAction
+
+    }
+}
+
+// MARK: - AuthContextProvider
+extension NotificationRequestsTableViewController: AuthContextProvider {
+    var authContext: AuthContext { viewModel.authContext }
+}
+
+extension NotificationRequestsTableViewController: NotificationRequestTableViewCellDelegate {
+    func acceptNotificationRequest(_ cell: NotificationRequestTableViewCell, notificationRequest: MastodonSDK.Mastodon.Entity.NotificationRequest) {
+
+        cell.acceptNotificationRequestActivityIndicatorView.isHidden = false
+        cell.acceptNotificationRequestActivityIndicatorView.startAnimating()
+        cell.acceptNotificationRequestButton.tintColor = .clear
+        cell.acceptNotificationRequestButton.setTitleColor(.clear, for: .normal)
+        cell.rejectNotificationRequestButton.isUserInteractionEnabled = false
+        cell.acceptNotificationRequestButton.isUserInteractionEnabled = false
+
+        Task { [weak self] in
+            guard let self else { return }
+            do {
+                try await acceptNotificationRequest(notificationRequest)
+            } catch {
+                cell.acceptNotificationRequestActivityIndicatorView.stopAnimating()
+                cell.acceptNotificationRequestButton.tintColor = .white
+                cell.acceptNotificationRequestButton.setTitleColor(.white, for: .normal)
+                cell.rejectNotificationRequestButton.isUserInteractionEnabled = true
+                cell.acceptNotificationRequestButton.isUserInteractionEnabled = true
+            }
+        }
+    }
+
+    private func acceptNotificationRequest(_ notificationRequest: MastodonSDK.Mastodon.Entity.NotificationRequest) async throws {
+        _ = try await context.apiService.acceptNotificationRequests(authenticationBox: authContext.mastodonAuthenticationBox,
+                                                                id: notificationRequest.id)
+
+        let requests = try await context.apiService.notificationRequests(authenticationBox: authContext.mastodonAuthenticationBox).value
+
+        NotificationCenter.default.post(name: .notificationFilteringChanged, object: nil)
+
+        if requests.count > 0 {
+            await MainActor.run { [weak self] in
+                guard let self else { return }
+                self.viewModel.requests = requests
+                var snapshot = NSDiffableDataSourceSnapshot<NotificationRequestsSection, NotificationRequestItem>()
+                snapshot.appendSections([.main])
+                snapshot.appendItems(self.viewModel.requests.compactMap { NotificationRequestItem.item($0) } )
+
+                self.dataSource?.apply(snapshot)
+            }
+        } else {
+            await MainActor.run { [weak self] in
```

Comment:
You could lift the `await MainActor.run { [weak self] in` out of the `if/else` to only have this once.